### PR TITLE
Remove the marketplace v1 and v2 APIs

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**

Removes the v1 and v2 API type definitions and they're references throughout the codebase. The CatalogSourceConfig and OperatorSource APIs have been deprecated and removed for a couple of releases now, and this controller is now responsible for simply ensuring the default set of CatalogSource specifications are correctly present or absent on the cluster depending on the configuration present in the OperatorHub cluster singleton type.

**Motivation for the change:**
 This removes these API type definitions entirely to simplify the codebase and improve the maintainability.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -14,7 +13,6 @@ reviewers:
   - timflannagan
 # approval == this is a good idea /approve
 approvers:
-  - ecordell
```
Comment (njhale) [inline OWNERS:17]: ???

File: pkg/controller/catalogsource/catalogsource_controller.go [hunk 2]
```diff
@@ -34,13 +33,12 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
...
 
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
-
 	c, err := controller.New("catalogsource-controller", mgr, controller.Options{Reconciler: r})
 	if err != nil {
 		return err
 	}
 
-	_, defaultCatalogsources := defaults.GetGlobalDefinitions()
+	defaultCatalogsources := defaults.GetGlobalCatalogSourceDefinitions()
```
Comment (anik120) [inline pkg/controller/catalogsource/catalogsource_controller.go:41]: nit: Is the whole `GetGlobalCatalogSourceDefinitions`really necessary? In fact early on in the code base we used `Get` a lot and that just stuck around as convention. `GlobalDefinitions` seems sufficient here.

**Review by anik120 — APPROVED** [review]
/lgtm

File: pkg/status/status.go [hunk 3]
```diff
@@ -90,9 +88,7 @@ func (r *reporter) setStatus(statusConditions []configv1.ClusterOperatorStatusCo
 	for _, statusCondition := range statusConditions {
 		r.setStatusCondition(statusCondition)
 	}
-
-	err = r.updateStatus(previousStatus)
-	if err != nil {
+	if err := r.updateStatus(previousStatus); err != nil {
```
Comment (kevinrizza) [inline pkg/status/status.go:91]: :)

**Review by kevinrizza — APPROVED** [review]
/approve

**Pull request merged**